### ISSUE TITLE
php-imap mail fails to detect MIME type in 2.2.x

### ISSUE THREAD

**Issue by lucasRolff** [barbushin/php-imap#199]
Hi,

We used php-imap to parse emails, recently the code was changed in the getMail function to something like:

```php
	public function getMailHeader($mailId) {
		$headersRaw = $this->imap('fetchheader', [$mailId, FT_UID]);
		$head = $this->imap('rfc822_parse_headers', $headersRaw, false);
```

It used to do:

```php
	public function getMail($mailId, $markAsSeen = true) {
		$headersRaw = imap_fetchheader($this->getImapStream(), $mailId, FT_UID);
		$head = imap_rfc822_parse_headers($headersRaw);
```

After the change, it results in following error:

```
Fatal error:  Uncaught PhpImap\Exception: IMAP method imap_rfc822_parse_headers() failed with error: Warning: message has unknown MIME version in php-imap-test/PhpImap/Mailbox.php:829
Stack trace:
#0 php-imap-test/PhpImap/Mailbox.php(509): PhpImap\Mailbox->imap('rfc822_parse_he...', Array, false)
#1 php-imap-test/PhpImap/Mailbox.php(575): PhpImap\Mailbox->getMailHeader(1)
```

The email headers from `$headersRaw` are as following:

```
MIME-Version: 1.0
Received: from AM5P192MB0195.EURP192.PROD.OUTLOOK.COM (10.175.13.23) by
 AM5P192MB0195.EURP192.PROD.OUTLOOK.COM (10.175.13.23) with Microsoft SMTP
 Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384_P256) id
 15.20.77.7 via Mailbox Transport; Sun, 8 Oct 2017 20:49:52 +0000
Received: from AM5P192MB0148.EURP192.PROD.OUTLOOK.COM (10.175.13.12) by
 AM5P192MB0195.EURP192.PROD.OUTLOOK.COM (10.175.13.23) with Microsoft SMTP
 Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384_P256) id
 15.20.77.7; Sun, 8 Oct 2017 20:49:52 +0000
Authentication-Results: example.com; dkim=none (message not signed)
 header.d=none;example.com; dmarc=none action=none
 header.from=example.com;
Received: from AM5P192MB0195.EURP192.PROD.OUTLOOK.COM (10.175.13.23) by
 AM5P192MB0148.EURP192.PROD.OUTLOOK.COM (10.175.13.12) with Microsoft SMTP
 Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384_P256) id
 15.20.77.7; Sun, 8 Oct 2017 20:49:52 +0000
Received: from AM5P192MB0195.EURP192.PROD.OUTLOOK.COM
 ([fe80::f5c3:bde:1eac:f96a]) by AM5P192MB0195.EURP192.PROD.OUTLOOK.COM
 ([fe80::f5c3:bde:1eac:f96a%18]) with mapi id 15.20.0077.020; Sun, 8 Oct 2017
 20:49:51 +0000
From: DOMAIN General 
To: DOMAIN Support 
Subject: Re: [Ticket ID: 233663] Lad os teste tickets nu
Thread-Topic: [Ticket ID: 233663] Lad os teste tickets nu
Thread-Index: AQHTQHaqbGiHRFwQ3UixNrnyHpVpoaLabQuA
Date: Sun, 8 Oct 2017 20:49:51 +0000
Message-ID: <834AAFB8-6A6B-46BF-85CF-7E1C420B59E5@example.com>
References: <0102015efdbe1b5a-ab7c03b0-805c-4165-bbd3-c7840884c371-000000@eu-west-1.amazonses.com>
In-Reply-To: <0102015efdbe1b5a-ab7c03b0-805c-4165-bbd3-c7840884c371-000000@eu-west-1.amazonses.com>
Accept-Language: en-US
Content-Language: en-GB
X-MS-Has-Attach: yes
X-MS-Exchange-Organization-SCL: -1
X-MS-TNEF-Correlator: <834AAFB8-6A6B-46BF-85CF-7E1C420B59E5@example.com>
X-MS-Exchange-Organization-MessageDirectionality: Originating
X-MS-Exchange-Organization-AuthSource: AM5P192MB0195.EURP192.PROD.OUTLOOK.COM
X-MS-Exchange-Organization-AuthAs: Internal
X-MS-Exchange-Organization-AuthMechanism: 04
X-Originating-IP: [2a02:a44e:xx:x:xx:xx:xx:xx]
X-MS-Exchange-Organization-Network-Message-Id: 7537f6be-48c9-4e45-bbc4-08d50e8e1f85
X-MS-PublicTrafficType: Email
X-Microsoft-Exchange-Diagnostics: 1;AM5P192MB0195;35:1Ebr4cjyVeDJFs2xnxKydxiUegibldPcUe4gRx6POpsufipCg7MFgkCbbt4thJtV3ThSb1wCOfXZaVM3ODxZhg==
X-MS-Exchange-Antispam-SRFA-Diagnostics: SSOS;
X-MS-Exchange-Organization-AVStamp-Service: 1.0
Return-Path: general@example.com
X-MS-Office365-Filtering-Correlation-Id: 7537f6be-48c9-4e45-bbc4-08d50e8e1f85
X-Microsoft-Antispam: UriScan:;BCL:0;PCL:0;RULEID:(22001)(2017030254152)(2017082002075)(2017052603199)(49563074)(201703131423075)(201702281549075);SRVR:AM5P192MB0148;
X-Microsoft-Exchange-Diagnostics: 1;AM5P192MB0148;3:JVm0v9e/XcqzmDb2b4JTOJwxRH7J5qZs4YPudZ4leuEPIi5M9Pmq5ubIE87MNY88SWg6NAGCBc+984tC386IDfpDWmYG6YspkhkLZxxvX+tb8dVFcyZkBA6/8AqLyMTxTAthzy2j4wlUPQcAi7BhaNMQa5NoSWCEvlOiRJ+eCjorKQ8ybcny/UfXX/z7GHubo0slADvXA97h7GBjuXXuXFzQhXzDLdzUGf99dc4b2ogbpEOI/NHcD6qLQfQ3M/Qw;25:gsK6Vsj038itcqKUWd1Y/ndok6MuxypvYdCtN9DUNNKn80hRNfWLnfeSt/rMW/DikOA5N1knGQrhJbrKNXl96QUFej3omcRdQBDMmgd77flJHDcaiIp+tTk54xWMicWhIky6HYSZL8v9z8fOJsEeQWb4UxDOs7hm7P3erFP3BG38/TukVnO6t5fJ+Rpm6QIvgFlsWH0Idt95vlQIVdpY/xIdAXwnuDTLFABAeX89W3g=;31:l+/2BPdrhl3QcvLAf9TyWFFcp48x7eCqZQ1m/fpRTgP5R4lri4EeokS9q4nR7fAKqfZcIETOyQrKvwsOBbxDEGVBVWGF3ygPs6l/JVGTPMcHC4Piis3scwkwFd5+slCJ;20:XmOZCron0RYxooocboLBUqagWClatFk7M4Hw+6ntfbAbHvYFmLtgmfn6ybcP7yJhRff3xaZZ0KkjmTldy94cOrlDIBmCXmmPyf1zIk9IpVkHMFAFWSgR/ZHJUyntbsuXrq7mxgQZ8wvFawlzP4gWGF5JNMSmcbTx30cXd7EhvUny9Ig28Dby8b5QLsdW5ek2v4jyfgFVDaA+Xb+T4irFpvqzYQNfHuX5HLJAdH2TAW7FX/pAFhCdFNKz7Q7ZPqWk
X-MS-TrafficTypeDiagnostic: AM5P192MB0148:|AM5P192MB0195:
X-Exchange-Antispam-Report-Test: UriScan:(21748063052155);
X-Exchange-Antispam-Report-CFA-Test: BCL:0;PCL:0;RULEID:(100000700101)(100105000095)(100000701101)(100105300095)(100000702101)(100105100095)(102415395)(9101524173)(2401047)(8121501046)(93006095)(93001095)(10201501046)(3002001)(100000703101)(100105400095)(201708071742011)(100000704101)(100105200095)(100000705101)(100105500095);SRVR:AM5P192MB0148;BCL:0;PCL:0;RULEID:(100000800101)(100110000095)(100000801101)(100110300095)(100000802101)(100110100095)(100000803101)(100110400095)(100000804101)(100110200095)(100000805101)(100110500095);SRVR:AM5P192MB0148;
X-Microsoft-Exchange-Diagnostics: 1;AM5P192MB0148;4:2DrrSVuKCblG34eadnEpxX/HHQqMLljSiAP1kpKrzMwy8wDpBrkCuznNz1827keiUZVnXqXO1bKmmk7xHQ1TP0M80SXP2vZtE+GYrZfbTQAClos7fREYg0KiQtjJAVl7Li9RVt+2Rm9sYO3krnI48VoLTIUZZiOYI5my9lkduNwDsmHgd0mlUB26HqHnJMFF;23:TPj9jsvcJX/xT2EYMqbYOtLjltmV86hC4itPcNiBQUwPx2mZ1zBiCiTE7lifCZKIWcuHU0dyAaQ3EZKKXIpzGjflXiWFRlp9WehLwx1BWQ7Ua4CPAqgWJsNTyW+N69pXaeZX3gB7+MWAsN0BUjep4Q==;6:XWQl73hJQw7GgndyfohFmNKJMgtk3/uZoZxnQ+A+3YAhI4U+ca1Qh3ivOOZckVp75yEjn9QHDEpJcdbnKq80ESruBmKZqIX2NVoo41bjyhooZ01FbAX5mR1w2Pnq5NOS8POiXk7+95HeDvYqFwu/vfWfi9q+XQQ8KKHLFSUR4KzygUdzUPHhEFIVzfdceSOg5NhPsMVwO7iasppOBVODdZof1WPNPLDhvxf3tAtoPO4=;5:jpi+I2oBsFe2asdKdvG9BNQyL0IZTy55NCCdIZtOygULF5+WrM4ipVNwGK5b3SscVBNLnVHufjJog7CnT44NHV9HQL7ccHqhDfpapYFEVX/Z/KA6JVqDgsjMbWnchTqnkkZJFJv8RuxRBuDrJgve/A==;24:S8nO98Of6pKjfEv6HkSG66AoRPYgyo1tRkp+YUQzQg4TVBOZaGQi7ZOvgFRkVTXTEIVEufynw7xCCu6qIHIRFBkPhADJ4F8WfZdUqWLyDqA=
X-Forefront-Antispam-Report: SFV:SKI;SFS:;DIR:INB;SFP:;SCL:-1;SRVR:AM5P192MB0148;H:AM5P192MB0195.EURP192.PROD.OUTLOOK.COM;FPR:;SPF:None;LANG:da;;SKIP:1;
SpamDiagnosticOutput: 1:0
X-Microsoft-Exchange-Diagnostics: 1;AM5P192MB0148;7:/htCwN2I+ep5zXmQ8d0FEktoF642TbTlZ4U/rJZ0cDMfw2ATTeRL5lJ79ZyhX52p/GtZJSpMeYRpXnH+mOh0capy2InCXESS9aOYYZ2ibrhjyYiVEtJESsg7hRl8ZSNC265vpvf99HWtzgrpJ8I1meC5j7so5bGaGsrdoWbRHtSnkhKLkC/R+/5BVPBGwnB6ltc3JU2rJA+Vvmvb+3I4RtaXQOGxmbvgl6KiUShI9no=
X-MS-Exchange-Organization-Recipient-P2-Type: To
X-MS-Exchange-CrossTenant-OriginalArrivalTime: 08 Oct 2017 20:49:51.9171
 (UTC)
X-MS-Exchange-CrossTenant-FromEntityHeader: Hosted
X-MS-Exchange-CrossTenant-Id: 3ffca99c-0e7e-499c-891d-188ba95e4793
X-MS-Exchange-Transport-CrossTenantHeadersStamped: AM5P192MB0148
X-Microsoft-Exchange-Diagnostics: 1;AM5P192MB0195;23:cr8JuZznvbd7RJ3j4yjjyJ2S10zRtHGajpgOh0o8aa5Typ05FJ4XHLy9I8rOxVlly/62c5DUlUyFcxhIox9HOQIxWUmPtVhpIGaXgsSMNU0vk0rp13iJitnwRvNMlsTpw4p3SLgcP4sLLxJFTtasbQ==
X-MS-Exchange-Transport-EndToEndLatency: 00:00:01.0312668
X-MS-Exchange-Processed-By-BccFoldering: 15.20.0077.017
X-Microsoft-Exchange-Diagnostics: 1;AM5P192MB0195;9:2u+bgwW61NGWPFZm+WZI5dYuFElZ1DWtmEXBqTOS9n+KRX8vPMivOamxK1iWmKeQvMP+SUNUQrse+b2ReOFW7F5d/pv8MaNCXPOJ/Ib/effNP04eJ9/vVz6Q3furPz6M
X-Microsoft-Exchange-Diagnostics: 1;AM5P192MB0195;27:i9z4nM5VrGgDD3y3VRfYkzLSnEwunemTczcyFfoupvESUYS2bYZkCmG3GtSRvWArrv2edal39THokO5IZfIucnIYHHi1PD9F0xXYtNKkImhoNVWtVPtKZmE98b16KO9p
Content-type: multipart/mixed;
	boundary="B_3590348072_1000824923"


```

As you see the MIME-Version is the very first header of the email.

**Comment by aymanrb** [comment]
Should be fixed in the new version v3.0.1

Issue: #201

**Comment by barbushin** [comment]
```
$head = $this->imap('rfc822_parse_headers', $headersRaw, false);
```
This code was released in 3.0.0, and fixed in 3.0.1
